pkgs/development/python-modules/jax/default.nix

```diff
@@ -17,97 +17,96 @@
+    # Not a big deal, this is how the JAX docs suggest running the test suite
+    # anyhow.
+    pytestFlagsArray = [
+      "--numprocesses=4"
+      "-W ignore::DeprecationWarning"
+      "tests/"
+    ];
 
-  disabledTests = [
-    # Exceeds tolerance when the machine is busy
-    "test_custom_linear_solve_aux"
-  ] ++ lib.optionals usingMKL [
-    # See
-    #  * https://github.com/google/jax/issues/9705
-    #  * https://discourse.nixos.org/t/getting-different-results-for-the-same-build-on-two-equally-configured-machines/17921
-    #  * https://github.com/NixOS/nixpkgs/issues/161960
-    "test_custom_linear_solve_cholesky"
-    "test_custom_root_with_aux"
-    "testEigvalsGrad_shape"
-  ];
+    disabledTests = [
+      # Exceeds tolerance when the machine is busy
+      "test_custom_linear_solve_aux"
+    ] ++ lib.optionals usingMKL [
+      # See
+      #  * https://github.com/google/jax/issues/9705
+      #  * https://discourse.nixos.org/t/getting-different-results-for-the-same-build-on-two-equally-configured-machines/17921
+      #  * https://github.com/NixOS/nixpkgs/issues/161960
+      "test_custom_linear_solve_cholesky"
+      "test_custom_root_with_aux"
+      "testEigvalsGrad_shape"
+    ];
 
-  # See https://github.com/google/jax/issues/11722. This is a temporary fix in
-  # order to unblock etils, and upgrading jax/jaxlib to the latest version. See
-  # https://github.com/NixOS/nixpkgs/issues/183173#issuecomment-1204074993.
-  disabledTestPaths = [
-    "tests/api_test.py"
-    "tests/core_test.py"
-    "tests/lax_numpy_indexing_test.py"
-    "tests/lax_numpy_test.py"
-    "tests/nn_test.py"
-    "tests/random_test.py"
-    "tests/sparse_test.py"
-  ];
+    # See https://github.com/google/jax/issues/11722. This is a temporary fix in
+    # order to unblock etils, and upgrading jax/jaxlib to the latest version. See
+    # https://github.com/NixOS/nixpkgs/issues/183173#issuecomment-1204074993.
+    disabledTestPaths = [
+      "tests/api_test.py"
+      "tests/core_test.py"
+      "tests/lax_numpy_indexing_test.py"
+      "tests/lax_numpy_test.py"
+      "tests/nn_test.py"
+      "tests/random_test.py"
+      "tests/sparse_test.py"
+    ];
 
-  pythonImportsCheck = [
-    "jax"
-  ];
+    pythonImportsCheck = [
+      "jax"
+    ];
 
-  meta = with lib; {
-    description = "Differentiate, compile, and transform Numpy code";
-    homepage = "https://github.com/google/jax";
-    license = licenses.asl20;
-    maintainers = with maintainers; [ samuela ];
+    passthru.tests = {
+      check = jax.overridePythonAttrs (_: { doCheck = true; });
+    };
+
+    meta = with lib; {
+      description = "Differentiate, compile, and transform Numpy code";
+      homepage = "https://github.com/google/jax";
+      license = licenses.asl20;
+      maintainers = with maintainers; [ samuela ];
+    };
   };
-}
+in
+jax
```

Comment:
Please use `let usingMKL = ...; in buildPythonPackage { ... }` instead of `let usingMKL = ...; jax = buildPythonPackage { ... }; in jax`

Reply:
Edit: I'm guessing that this is necessary for `passthru.tests`... I would still much prefer to break the dependency cycle at the etils level since that package is much, much smaller than JAX.

Reply:
Also passthru.tests does not get run as reliably as the standard test suite in most tooling

Reply:
Should be resolved now (see https://github.com/NixOS/nixpkgs/pull/183051#issuecomment-1224954563)